File: src/main/java/com/github/ybqdren/lc/lc34.java
package com.github.ybqdren.lc;

/**
 * created by ybqdren
 */
public class lc34 {
    public int[] searchRange(int[] nums, int target) {
        int[] ans = new int[]{-1,-1};
        int left = binarySearch(nums , target , true);
        int right = binarySearch(nums ,target , false);
        if(left ==-1 || right == -1){
            return ans;
        }

        ans[0] = left;
        ans[1] = right;
        return ans;
    }


    public int binarySearch(int[] nums , int target , boolean isLow){
        int left = 0 , right = nums.length-1;
        int index = -1;
        while(left <= right){
            int mid = (right - left)/2 + left;
            if(nums[mid] > target){
                right = mid - 1;
            }else if(nums[mid] < target){
                left = mid + 1;
            }else{
                if(isLow){
                    return mid;
                }else{
                    index = mid;
                }
                left = mid + 1;
            }
        }
        return index;
    }
}

File: src/main/java/com/github/ybqdren/lc/lc34.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        l , r = 0 ,len(nums) - 1
        while l <= r:
            mid = l + (r - l) // 2
            if nums[mid] == target:
                i = j = mid
                while i < len(nums) - 1 and nums[i] == target:
                    i+=1
                while j >= 0 and nums[j] == target:
                    j -= 1
                return  [j + 1 ,i - 1]
            elif nums[mid] > target:
                r = mid - 1
            else:
                l = mid + 1
        return  [-1 , -1]


solution =  Solution()

nums = [5,7,7,8,8,10]
target = 8
solution.searchRange(nums , target)